core/src/main/java/org/opensearch/sql/data/model/ExprDateValue.java

```diff
@@ -25,20 +32,44 @@
 public class ExprDateValue extends AbstractExprValue {
 
   private final LocalDate date;
+  private String datePattern;
+  // List of cleaned patterns
+  private static final List<String> CLEANED_PATTERNS = Arrays.stream(DATE_TIME_FORMATTER_PATTERNS_NANOS_OPTIONAL.split("\\]\\["))
+          .map(pattern -> pattern.replace("[", "").replace("]", ""))
+          .collect(Collectors.toList());
```

Comment:
It was a bit difficult to understand what this is doing. It would be helpful for reading if it is extracted as method like `extractDateTimePatterns`.
I would consider generating pattern from list instead of parsing from string. (That would be more error prune)

Reply:
Sure, I've converted the patterns as list and do not need this cleaning part anymore.